modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/UserCardTag.java

```diff
@@ -0,0 +1,522 @@
+		if (_selected == null) {
+			if (_userCard != null) {
+				return _userCard.isSelected();
+			}
+
+			return false;
+		}
+
+		return _selected;
+	}
+
+	public void setActionDropdownItems(List<DropdownItem> actionDropdownItems) {
+		_actionDropdownItems = actionDropdownItems;
+	}
+
+	public void setDisabled(Boolean disabled) {
+		_disabled = disabled;
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
+	 */
+	@Deprecated
+	public void setGroupName(String groupName) {
+		_groupName = groupName;
+	}
+
+	public void setHref(String href) {
+		_href = href;
+	}
+
+	public void setIcon(String icon) {
+		_icon = icon;
+	}
+
+	public void setImageAlt(String imageAlt) {
+		_imageAlt = imageAlt;
+	}
+
+	public void setImageSrc(String imageSrc) {
+		_imageSrc = imageSrc;
+	}
+
+	public void setInputName(String inputName) {
+		_inputName = inputName;
+	}
+
+	public void setInputValue(String inputValue) {
+		_inputValue = inputValue;
+	}
+
+	public void setLabels(List<LabelItem> labels) {
+		_labels = labels;
+	}
+
+	public void setName(String name) {
+		_name = name;
+	}
+
+	public void setSelectable(Boolean selectable) {
+		_selectable = selectable;
+	}
+
+	public void setSelected(Boolean selected) {
+		_selected = selected;
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
+	 */
+	@Deprecated
+	public void setSpritemap(String spritemap) {
+		_spritemap = spritemap;
+	}
+
+	public void setSubtitle(String subtitle) {
+		_subtitle = subtitle;
+	}
+
+	public void setUserCard(UserCard userCard) {
+		_userCard = userCard;
+	}
+
+	public void setUserColorClass(String userColorClass) {
+		_userColorClass = userColorClass;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_actionDropdownItems = null;
+		_disabled = null;
+		_groupName = null;
+		_href = null;
+		_icon = null;
+		_imageAlt = null;
+		_imageSrc = null;
+		_inputName = null;
+		_inputValue = null;
+		_labels = null;
+		_name = null;
+		_selectable = null;
+		_selected = null;
+		_spritemap = null;
+		_subtitle = null;
+		_userCard = null;
+		_userColorClass = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/UserCard";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("actions", getActionDropdownItems());
+		props.put("description", getSubtitle());
+		props.put("disabled", isDisabled());
+		props.put("href", getHref());
+		props.put("id", getId());
+		props.put("inputName", getInputName());
+		props.put("inputValue", getInputValue());
+		props.put("userImageAlt", getImageAlt());
+		props.put("userImageSrc", getImageSrc());
+		props.put("userSymbol", getIcon());
+		props.put("labels", getLabels());
+		props.put("name", getName());
+		props.put("selectable", isSelectable());
+		props.put("selected", isSelected());
+		props.put("userDisplayType", getUserColorClass());
+
+		return super.prepareProps(props);
```

Comment:
We're applying the same logic that exists in the Clay component here so the server side render uses the same one. We don't need to replicate the same for client to send it to the Clay component, which already have that logic

Reply:
Or do you mean that we should NOT pass those cssClasses to the clay component?

Reply:
Yes, I don't get what both options you mean

Reply:
Just to clarify  how this works:

`BaseContainerTag.processStartTag` renders the main element, adding to it the result of `processCssClasses`.

`BaseContainerTag.processCssClasses` gets all `cssClasses` coming from the method that overwrite it and add to them the `cssClass` attribute (which the user can pass to the tag). Those classes are returned and renderer as explained above in the `processStartTag` method.

`processCssClasses` is meant to be a place to **replicate the same logic for css classes located in the correspondent Clay Coomponent** . Nothing else.

So, if the Tag is Hydrated, the wrapper component receives `cssClass`, which is the attribute the user can specify and pass it as the `className` value of the Clay component.

We don't need to pass to the component the `processCssClass` result because that's something already present in the react component.

Reply:
Yeah, this boils down to the discussion of using Clay 2 vs Clay 3 markup & logic. 

If you are set to recreate Clay 2 markup, `processCssClasses` is a useful tool to achieve the same logic and markup in Clay 3, just like it was in achieving the same logic and markup in server side rendering.

If you are set to use Clay 3 markup, you don't care about the logic and classes for client side, they are already there in the component.

Reply:
I think it should be enough with `cssClass` since the others are indeed part of the React component.

Forget about Clay2. Let's focus on advancing this and updating the usages. Let's leave Clay2 behind as fast as we can!

Reply:
> Forget about Clay2. Let's focus on advancing this and updating the usages. Let's leave Clay2 behind as fast as we can!

@jbalsas Should we document all the breaking changes?

To quote myself:
>  What if a customer used v2 markup for custom styles? Upgrade to new Liferay version could break his component. We are not creating new tag here, we are switching implementation, so this is a very real and likely scenario.